R/mag.R

```diff
@@ -54,10 +53,10 @@ mag <- function (data, n = 60L, dt0 = NULL, inter_gap = 45, tz = "") {
       n <- dt0
     }
 
-    idx = seq(1, ncol(data_ip[[1]]), by = round(n/data_ip[[3]]))
-    idx_gl = as.vector(t(data_ip[[1]][, idx]))
-    mag = sum(abs(diff(idx_gl)), na.rm = TRUE)/
-      (length(na.omit(idx_gl))*n/60)
+    diffs = diff(as.vector(t(data_ip[[1]])))
+    total_time = (length(na.omit(diffs))*dt0)/n
```

Comment:
if total_time is in hours, then n should always be 60. if n is less than 60, then I am not sure the logic is appropriate. The original code was specifically taking differences over an interval controlled by n. THe new code will always take consecutive differences so this control is eliminated. Better to stick to the original code, but modify for NA